skbio/diversity/_driver.py

```diff
@@ -0,0 +1,304 @@
+     * ``tree`` is rooted
+     * ``tree`` has more than one node
+     * all nodes in ``tree`` except for the root node have branch lengths
+     * all tip names in ``tree`` are unique
+     * all ``otu_ids`` correspond to tip names in ``tree``
+
+    """
+    metric_map = _get_alpha_diversity_metric_map()
+
+    if validate:
+        counts = _validate_counts_matrix(counts, ids=ids)
+
+    if metric == 'faith_pd':
+        otu_ids, tree, kwargs = _get_phylogenetic_kwargs(counts, **kwargs)
+        counts_by_node, branch_lengths = _setup_faith_pd(
+            counts, otu_ids, tree, validate, single_sample=False)
+        counts = counts_by_node
+        metric = partial(_faith_pd, branch_lengths=branch_lengths)
+    elif callable(metric):
+        metric = partial(metric, **kwargs)
+    elif metric in metric_map:
+        metric = partial(metric_map[metric], **kwargs)
+    else:
+        raise ValueError('Unknown metric provided: %r.' % metric)
+
+    results = [metric(c) for c in counts]
+    return pd.Series(results, index=ids)
+
+
+@experimental(as_of="0.4.0")
+def beta_diversity(metric, counts, ids=None, validate=True, **kwargs):
+    """Compute distances between all pairs of samples
+
+    Parameters
+    ----------
+    metric : str, callable
+        The pairwise distance function to apply. See the scipy ``pdist`` docs
+        and the scikit-bio functions linked under *See Also* for available
+        metrics. Passing metrics as a strings is preferable as this often
+        results in an optimized version of the metric being used.
+    counts : 2D array_like of ints or floats
+        Matrix containing count/abundance data where each row contains counts
+        of OTUs in a given sample.
+    ids : iterable of strs, optional
+        Identifiers for each sample in ``counts``.
+    validate: bool, optional
+        If `False`, validation of the input won't be performed. This step can
+        be slow, so if validation is run elsewhere it can be disabled here.
+        However, invalid input data can lead to invalid results or error
+        messages that are hard to interpret, so this step should not be
+        bypassed if you're not certain that your input data are valid. See
+        Notes for the description of what validation entails so you can
+        determine if you can safely disable validation.
+    kwargs : kwargs, optional
+        Metric-specific parameters.
+
+    Returns
+    -------
+    skbio.DistanceMatrix
+        Distances between all pairs of samples (i.e., rows). The number of
+        row and columns will be equal to the number of rows in ``counts``.
+
+    Raises
+    ------
+    ValueError, MissingNodeError, DuplicateNodeError
+        If validation fails (see description of validation in Notes). Exact
+        error will depend on what was invalid.
+    TypeError
+        If invalid method-specific parameters are provided.
+
+    See Also
+    --------
+    unweighted_unifrac
```

Comment:
use `skbio.diversity.beta` instead of `unweighted_unifrac` and `weighted_unifrac`